src/north/north-connector.js

```diff
@@ -243,23 +242,20 @@ class NorthConnector {
       return
     }
 
-    const fileToSend = await this.fileCache.retrieveFileFromCache()
-    if (!fileToSend) {
-      this.logger.trace('No file to send in the cache database.')
-      this.resetFilesTimeout(this.caching.sendInterval)
-      return
-    }
-    this.logger.trace(`File to send: "${fileToSend.path}".`)
-
+    let fileToSend
     try {
-      await fs.stat(fileToSend.path)
+      fileToSend = await this.fileCache.retrieveFileFromCache()
+      if (!fileToSend) {
+        this.logger.trace('No file to send in the cache folder.')
+        this.resetFilesTimeout(this.caching.sendInterval)
+        return
+      }
     } catch (error) {
-      // File in cache does not exist on filesystem
-      await this.fileCache.removeFileFromCache(fileToSend.path, false)
-      this.logger.error(`File "${fileToSend.path}" not found! The file has been removed from the cache.`)
-      this.resetFilesTimeout(this.caching.sendInterval)
+      // Error triggered when a file is being written (from a South) during the retrieveFileFromCache call
+      this.logger.error(error)
```

Comment:
is this catch block useful can an error be thrown before?

Reply:
fs.readdir can throw an error in the retrieveFileFromCache method if the folder does not exist. I'm gonna remove this logger.error and place the try catch in retrieveFileFromCache method around readdir